polkadot-parachains/integritee-runtime/src/weights/pallet_xcm.rs

```diff
@@ -0,0 +1,259 @@
+// --steps=50
+// --repeat=20
+// --pallet=pallet_xcm
+// --extrinsic=*
+// --execution=wasm
+// --wasm-execution=compiled
+// --heap-pages=4096
+// --output=./polkadot-parachains/integritee-runtime/src/weights/pallet_xcm.rs
+
+#![cfg_attr(rustfmt, rustfmt_skip)]
+#![allow(unused_parens)]
+#![allow(unused_imports)]
+
+use frame_support::{traits::Get, weights::Weight};
+use sp_std::marker::PhantomData;
+
+/// Weight functions for `pallet_xcm`.
+pub struct WeightInfo<T>(PhantomData<T>);
+impl<T: frame_system::Config> pallet_xcm::WeightInfo for WeightInfo<T> {
+	/// Storage: PolkadotXcm SupportedVersion (r:1 w:0)
+	/// Proof Skipped: PolkadotXcm SupportedVersion (max_values: None, max_size: None, mode: Measured)
+	/// Storage: PolkadotXcm VersionDiscoveryQueue (r:1 w:1)
+	/// Proof Skipped: PolkadotXcm VersionDiscoveryQueue (max_values: Some(1), max_size: None, mode: Measured)
+	/// Storage: PolkadotXcm SafeXcmVersion (r:1 w:0)
+	/// Proof Skipped: PolkadotXcm SafeXcmVersion (max_values: Some(1), max_size: None, mode: Measured)
+	/// Storage: ParachainSystem HostConfiguration (r:1 w:0)
+	/// Proof Skipped: ParachainSystem HostConfiguration (max_values: Some(1), max_size: None, mode: Measured)
+	/// Storage: ParachainSystem PendingUpwardMessages (r:1 w:1)
+	/// Proof Skipped: ParachainSystem PendingUpwardMessages (max_values: Some(1), max_size: None, mode: Measured)
+	fn send() -> Weight {
+		// Proof Size summary in bytes:
+		//  Measured:  `111`
+		//  Estimated: `5010`
+		// Minimum execution time: 52_000 nanoseconds.
+		Weight::from_ref_time(55_900_000)
+			.saturating_add(Weight::from_proof_size(5010))
+			.saturating_add(T::DbWeight::get().reads(5))
+			.saturating_add(T::DbWeight::get().writes(2))
+	}
+	/// Storage: Benchmark Override (r:0 w:0)
+	/// Proof Skipped: Benchmark Override (max_values: None, max_size: None, mode: Measured)
+	fn teleport_assets() -> Weight {
+		// Proof Size summary in bytes:
+		//  Measured:  `0`
+		//  Estimated: `0`
+		// Minimum execution time: 18_446_744_073_709_551 nanoseconds.
+		Weight::from_ref_time(18_446_744_073_709_551_000)
+			.saturating_add(Weight::from_proof_size(0))
+	}
+	/// Storage: ParachainInfo ParachainId (r:1 w:0)
+	/// Proof: ParachainInfo ParachainId (max_values: Some(1), max_size: Some(4), added: 499, mode: MaxEncodedLen)
+	fn reserve_transfer_assets() -> Weight {
+		// Proof Size summary in bytes:
+		//  Measured:  `32`
+		//  Estimated: `499`
+		// Minimum execution time: 42_600 nanoseconds.
+		Weight::from_ref_time(43_600_000)
+			.saturating_add(Weight::from_proof_size(499))
+			.saturating_add(T::DbWeight::get().reads(1))
+	}
+	/// Storage: Benchmark Override (r:0 w:0)
+	/// Proof Skipped: Benchmark Override (max_values: None, max_size: None, mode: Measured)
+	fn execute() -> Weight {
+		// Proof Size summary in bytes:
+		//  Measured:  `0`
+		//  Estimated: `0`
+		// Minimum execution time: 18_446_744_073_709_551 nanoseconds.
+		Weight::from_ref_time(18_446_744_073_709_551_000)
```

Comment:
this shouldn't have slipped our reviews. very funny execution time indeed. this most likely breaks receiving XCM messages and makes everything overweight. CC @clangenb

Reply:
Hmm, interesting. Funnily, moonbeam has the same weird weight for teleport assets, but their `execute()` weight seems fine:  https://github.com/moonbeam-foundation/moonbeam/blob/d4fdae0cf795ebb48b7a2a9c2167c7dae56e782f/runtime/common/src/weights/pallet_xcm.rs#L75

Reply:
fixing this doesn't solve our problem, unfortunately. but needs fixing anyway